### PR TITLE
fix: skip custom check monitors missing options.groupby at sync time

### PR DESCRIPTION
**Pull request by michael-richey**
## Summary

- Custom check (service check) monitors were failing sync with `400: "A grouping must be specified for custom checks"`
- Root cause: source monitors have an empty or missing `options.groupby`; the destination API requires at least one entry for this monitor type
- Fix: detect the condition in `pre_resource_action_hook` on `Monitors`, skip at sync time with actionable message

## Changes

- `datadog_sync/model/monitors.py`: add `pre_resource_action_hook` that checks service check monitors for a non-empty `options.groupby`
- `tests/unit/test_monitors.py`: 7 unit tests covering all field/type/null edge cases

## Test plan

- [x] `pytest tests/unit/` — all unit tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: datadog_sync/model/monitors.py [hunk 1]
```diff
@@ -73,7 +73,15 @@ async def import_resource(self, _id: Optional[str] = None, resource: Optional[Di
...
 
     async def pre_resource_action_hook(self, _id, resource: Dict) -> None:
-        pass
+        if resource.get("type") == "service check":
+            groupby = (resource.get("options") or {}).get("groupby", [])
+            if not groupby:
+                raise SkipResource(
+                    _id,
+                    self.resource_type,
+                    "Custom check monitor requires at least one group-by. "
```
Comment (heyronhay) [inline datadog_sync/model/monitors.py]: I wonder if we should emit a common message for this kind of thing, like "DEPRECATED RESOURCE SETTING" or something?

**Review by heyronhay — DISMISSED** [review]

File: datadog_sync/model/monitors.py (hunk 1 above)
Comment (michael-richey) [reply datadog_sync/model/monitors.py]: Yeah, I was even thinking a totally different class. Not Successful, Skipped, or Failed. Skipping is more for resources that don't need updating. It's actually closer to Failed, but I like the idea of Deprecated.

Comment (heyronhay) [reply datadog_sync/model/monitors.py]: or "Needs Update"? Something along those lines. Makes sense to put it in a different category. "Cannot Sync"?

**Review by heyronhay — APPROVED** [review]
